### PR TITLE
Optimize Rename function logic to reduce the number of REST API calls.

### PR DESCRIPTION
**Pull request by syeleti-msft**
**Rename on FNS:**
API calls: There are 6 REST api calls assosiated with this operation.

1. GetAttr(src)---------list call (filesystem call)
2. GetAttr(dst)---------list call (filesystem call)
3. GetBlobproperties call(src) ------------(rename function)
4. copyblob(src->dst)----------(rename function)
5. deleteblob(src)--------------(rename function)
6. GetAttr(dst)---------list call (filesystem call)

3 is an extra call, removed it from the code

**Rename on HNS:**
API Calls: There are 4 REST api calls assosiated with this operation.

1. GetBlobProperties on Dst
2. GetBlobProperties on Source
3. RenamePathFile
4. GetBlobProperties on Dst

### CODE REVIEW

**Review by vibhansa-msft — APPROVED** [review]

File: component/azstorage/block_blob.go [hunk 1]
```diff
@@ -404,13 +391,25 @@ func (bb *BlockBlob) RenameDirectory(source string, target string) error {
...
+		serr := storeBlobErrToErr(err)
+		if serr == ErrFileNotFound && srcDirPresent {
+			return nil
+		} else if serr == ErrFileNotFound {
+			log.Err("BlockBlob::RenameDirectory : %s marker blob does not exist and Src Directory doesn't Exist", source)
+			return syscall.ENOENT
+		} else {
+			log.Err("BlockBlob::RenameDirectory : Failed to get source directory marker blob properties for %s [%s]", source, err.Error())
+			return err
+		}
```
Comment (jainakanksha-msft) [inline component/azstorage/block_blob.go:409]: > 
>if serr == ErrFileNotFound && srcDirPresent {
			return nil
		} else if serr == ErrFileNotFound {
			log.Err("BlockBlob::RenameDirectory : %s marker blob does not exist and Src Directory doesn't Exist", source)
			return syscall.ENOENT
		} else {
			log.Err("BlockBlob::RenameDirectory : Failed to get source directory marker blob properties for %s [%s]", source, err.Error())
			return err
		}
>


Simplified code can be
`
if serr == ErrFileNotFound {
		if srcDirPresent {
			return nil
		}
		log.Err("BlockBlob::RenameDirectory: Marker blob %s and source directory do not exist.", source)
		return syscall.ENOENT
	}

	// Other errors when getting blob properties
	log.Err("BlockBlob::RenameDirectory: Failed to get properties for marker blob %s: [%s]", source, err.Error())
	return err
`

File: component/azstorage/block_blob.go [hunk 2]
```diff
@@ -404,13 +391,25 @@ func (bb *BlockBlob) RenameDirectory(source string, target string) error {
 		}
 	}
 
-	err := bb.RenameFile(source, target)
-	// check if the marker blob for source directory does not exist but
-	// blobs were present in it, which were renamed earlier
-	if err == syscall.ENOENT && srcDirPresent {
-		err = nil
+	// To rename source marker blob check its properties before calling rename on it.
+	blobClient := bb.Container.NewBlockBlobClient(filepath.Join(bb.Config.prefixPath, source))
```
Comment (jainakanksha-msft) [inline component/azstorage/block_blob.go:395]: Please also add update test cases of modified conditions.

Comment (syeleti-msft) [reply component/azstorage/block_blob.go:402]: Tests were already been covered.

File: component/azstorage/block_blob.go (hunk 1 above)
Comment (syeleti-msft) [reply component/azstorage/block_blob.go:417]: Modified the Code

**Review by vibhansa-msft — APPROVED** [review]

**Review by ashruti-msft — APPROVED** [review]

**Review by jainakanksha-msft — APPROVED** [review]
LGTM

**Pull request merged**